pkgs/development/python-modules/click/default.nix

```diff
@@ -1,20 +1,33 @@
-{ lib, buildPythonPackage, fetchPypi, locale, pytestCheckHook }:
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, importlib-metadata
+, locale
+, pytestCheckHook
+}:
 
 buildPythonPackage rec {
   pname = "click";
-  version = "7.1.2";
+  version = "8.0.1";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "d2b5255c7c6349bc1bd1e59e08cd12acbbd63ce649f2588755783aa94dfb6b1a";
+    sha256 = "0ymdyf37acq4qxh038q0xx44qgj6y2kf0jd0ivvix6qij88w214c";
```

Comment:
It doesn't look like you're using the scraped/published SHA256's from pypi.org: e.g. https://pypi.org/project/click/8.0.1/#copy-hash-modal-d0461d48-855f-4668-bdc8-8e742a09f958. Would you be able to use those instead? They should evaluate to the same, I guess Nix's standard representation is a little different.

Reasons to use the published SHA's, IMHO:
* **Users can more readily validate these against PyPi and check the source integrity**
* Maybe something to do with making it scraping-friendly (both for updating nixpkgs & any downstream).

Reply:
Actually used `nix-update`, question is: Are you going to make the case for everyone touching these packages? Because there is no such policy in the python section. Maybe there should be, but then it should be discussed at a policy level.

Reply:
I would probably make that case, but I don't feel strongly enough about it to hold up this particular PR.

Whether there's a policy or not, my de facto understanding was that ``fetchPypi`` sources use the PyPi Sha256, based on many auto-updates that it seems @FRidh does. I'm fine with it as-is for this PR

Reply:
Those hashes are not base64 encoded and there is the possibility that nix calculates hashes different than pypi does. Doesn't matter in the end.